### PR TITLE
Adds a new event - Solargrubs!

### PR DESCRIPTION
**Pull request by Ketrai**
Changelog !!SPOILER ALERT!!
- Adds a new event similar to spiderinfestation (spawns around 8 grubs)
- adds solargrubs, a neutral mob capable of eating people, and latch onto cables/tiles with cables, sinking power from apc's on the grid. These mobs are quite beefy, have resistance to energy weapons, and deal burn damage through a special toxin when attacking. These mobs should encourage engineers to use sub-stations and give them more to do. they can check the power grid to locate grubs. Taking down a grub shouldn't be too easy, making it more of a job where any department can help. (you can easily 1v1 them with a decent weapon)
- adds meat and chemicals for the grubs
- adds exiting secret new chemicals
- adds sprites for well, the meat and grubs, duh. (borrowed mob sprite from /TG/)
- allows grubs to have a chance to come out of a predator crate.

Expecting to expand this mob in the future.

This update has been tested extensively

### CODE REVIEW

**Comment by Ketrai** [comment]
travis u wot

**Comment by Ketrai** [comment]
groaaan

**Comment by Ketrai** [comment]
There we go, travis was being travis. YAY! EVERYTHING WORKS!

File: code/game/objects/effects/grubs_vr.dm [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/obj/effect/spider/grub //travis is a butt!!!
```
Comment (SpadesNeil) [inline code/game/objects/effects/grubs_vr.dm]: You're inheriting the Spider effect but appear to have mostly just copied the Spider effect. I don't understand.

File: code/modules/reagents/reagent_containers/food/snacks_vr.dm [hunk 2]
```diff
@@ -190,6 +190,19 @@
 	reagents.add_reagent("protein", 4)
 	bitesize = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/grubmeat
```
Comment (SpadesNeil) [inline code/modules/reagents/reagent_containers/food/snacks_vr.dm]: Why is this _not_ a subclass of meat?

Comment (Ketrai) [reply code/modules/reagents/reagent_containers/food/snacks_vr.dm]: I'm planning on having grub meat used for 3 unique dishes. as far as I can tell if I make this a child of the meat code it'll be usable for all meat, I'm not a fan of that. xenomeat has its own seperate meat in the code

File: code/game/objects/effects/grubs_vr.dm (hunk 1 above)
Comment (Ketrai) [reply code/game/objects/effects/grubs_vr.dm]: Will look at this, see if I can cut down the code directly copied

Comment (Ketrai) [reply code/game/objects/effects/grubs_vr.dm]: going to look into this.

**Comment by Ketrai** [comment]
Don't merge, I messed something up cutting code

**Comment by Ketrai** [comment]
Hm, cutting code messes up the event.

**Comment by chaoko99** [comment]
Should probably squash this?

**Comment by Ketrai** [comment]
squash this? wha?

**Comment by Ketrai** [comment]
There, tested it, code is now significantly shorter, everything should be in order -v-

**Comment by SpadesNeil** [comment]
I'm fucking dumb and merged this when I didn't mean to do I probably just ruined all your work. <_>

Yea no we're not merging this until its problems are fixed and the meat recipes don't work like you think they work.

Spider meat recipes work with xeno meat because spider meat is a _subtype_ of xeno meat. A recipe that calls for _specifically_ spider meat won't work with xeno meat.

**Pull request merged**